test-workloads/src/bin/tellus.rs

```diff
@@ -954,11 +979,26 @@ extern "C" fn kernel_init(hart_id: u64, fdt_addr: u64) {
                                 }
                             }
                         }
-                        addr if !blocked => {
+                        addr if !fatal => {
                             // Fault in the page.
                             if zero_pages_added >= NUM_CONVERTED_ZERO_PAGES {
                                 panic!("Ran out of pages to fault in");
                             }
+
+                            if unshared_gpa_range
+                                .as_ref()
+                                .filter(|r| r.contains(&addr))
+                                .is_some()
+                                && unshared_spa_range.is_none()
+                            {
+                                unshared_spa_range = Some(Range {
+                                    start: zero_pages_base + zero_pages_added * PAGE_SIZE_4K,
+                                    end: zero_pages_base
+                                        + zero_pages_added * PAGE_SIZE_4K
+                                        + PAGE_SIZE_4K,
+                                });
+                            }
```

Comment:
Can you just add a small comment for this part explaining why we are saving unshared_spa_range here.
As per my understanding this is so that we can check if we need a fence or not and run the if block on line `810`. Right?
Seemed confusing to me at the start.

Reply:
Yes you understood that well. Of course I can add a comment to make things clearer.